warn.go

```diff
@@ -0,0 +1,93 @@
+package main
```

Comment:
I think it's too much for only one occurrence of `WarnTrivial`. Can we just write some structrure like:
```
type warningCollector struct {
    check bool
    warnings []string
}
```
And use just printf for trivial case.